packages/fxa-auth-server/lib/routes/emails.js

```diff
@@ -152,7 +152,10 @@ module.exports = (
             // of accounts with invalid email addresses. These
             // can never be verified, so the best we can do is
             // to delete them so the browser will stop polling.
-            if (!validators.isValidEmailAddress(sessionToken.email)) {
+            if (
+              !validators.isValidEmailAddress(sessionToken.email) &&
+              !(await stripeHelper.hasActiveSubscription(sessionToken.uid))
+            ) {
```

Comment:
I wasn't sure if we wanted to add the check here, but I did for thoroughness.

IIUC from the comment above, we would not in practice expect someone with an invalid email (and therefore older account) to also have an active subscription, since  prior to [the epic for this patch](https://mozilla-hub.atlassian.net/browse/FXA-3486), only verified and authenticated users could create a subscription.

Reply:
Can't hurt to add this here.